modules/nvidia_plugin/src/cuda/graph.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2021-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "graph.hpp"
+#include <ie_common.h>
+#include <fmt/format.h>
+
+namespace CUDA {
+
+Graph::Graph(unsigned int flags) :
+        Graph { createNativeWithFlags(flags) } {
+}
+
+Graph::Graph(cudaGraph_t graph) :
+        Handle { createFromNative, cudaGraphDestroy, graph } {
+}
+
+cudaError_t Graph::createFromNative(cudaGraph_t *pGraph, const cudaGraph_t anotherGraph) {
+    *pGraph = anotherGraph;
+    return cudaSuccess;
+}
+
+cudaGraph_t Graph::createNativeWithFlags(unsigned int flags) {
+    cudaGraph_t g;
+    throwIfError(cudaGraphCreate(&g, flags));
+    return g;
+}
+
+// clang-format off
+GraphExec::GraphExec(const Graph &g)
+#if !defined(NDEBUG)
+try
+#endif
+:
+Handle(cudaGraphInstantiate, cudaGraphExecDestroy, g.get(), static_cast<cudaGraphNode_t*>(nullptr),
+#if !defined(NDEBUG)
+       errorMsg_, kErrorStringLen)
+#else
+       static_cast<char*>(nullptr), 0ul)
+#endif
+{
+}
+#if !defined(NDEBUG)
+catch (std::exception &e) {
+    throw InferenceEngine::GeneralError { fmt::format("{}: {}", e.what(), errorMsg_) };
```

Comment:
> Use `ov::Exception` here

This exception type was choosen to keep consistency with other NVIDIA plugin components which throw `InferenceEngine::GeneralError` exception https://github.com/openvinotoolkit/openvino_contrib/blob/master/modules/nvidia_plugin/src/error.cpp#LL23C5-L23C19. I agree that exception type should be updated with a new one, but across all NVIDIA Plugin components. I'd propose to update exception type for all NVIDIA plugin components in a separate PR, but keep `InferenceEngine::GeneralError` in this PR for consitency. Otherwise it implies revising all NVIDIA Plugin codebase and updating exception types which is out of scope of this PR.

Reply:
@ytorzuk-altran you can use `throwIEException`, it is content modified in https://github.com/openvinotoolkit/openvino_contrib/pull/656/

Reply:
`throwIEException` adds additional details about exception location which previously added in the caught exception. `throwIEException` is not used here to avoid exception location details duplication.